pkg/api/roles.go

```diff
@@ -44,145 +44,157 @@ var (
+			Version:     2,
+			Name:        "fixed:datasources:reader",
+			DisplayName: "Data source reader",
+			Description: "Read and query all datasources.",
+			Permissions: []accesscontrol.Permission{
+				{
+					Action: ActionDatasourcesRead,
+					Scope:  ScopeDatasourcesAll,
+				},
+				{
+					Action: ActionDatasourcesQuery,
+					Scope:  ScopeDatasourcesAll,
 				},
 			},
-			Grants: []string{string(models.ROLE_ADMIN)},
 		},
-		{
-			Role: accesscontrol.RoleDTO{
-				Version:     2,
-				Name:        "fixed:datasources:id:viewer",
-				Description: "Gives access to read datasources ID",
-				Permissions: []accesscontrol.Permission{
-					{
-						Action: ActionDatasourcesIDRead,
-						Scope:  ScopeDatasourcesAll,
-					},
+		Grants: []string{string(models.ROLE_ADMIN)},
+	}
+
+	datasourcesWriterRole := accesscontrol.RoleRegistration{
+		Role: accesscontrol.RoleDTO{
+			Version:     2,
+			Name:        "fixed:datasources:writer",
+			DisplayName: "Data source writer",
+			Description: "Create, update, delete, read, or query data sources.",
+			Permissions: accesscontrol.ConcatPermissions(datasourcesReaderRole.Role.Permissions, []accesscontrol.Permission{
+				{
+					Action: ActionDatasourcesWrite,
+					Scope:  ScopeDatasourcesAll,
 				},
-			},
-			Grants: []string{string(models.ROLE_VIEWER)},
+				{
+					Action: ActionDatasourcesCreate,
+				},
+				{
+					Action: ActionDatasourcesDelete,
+					Scope:  ScopeDatasourcesAll,
+				},
+			}),
 		},
-		{
-			Role: accesscontrol.RoleDTO{
-				Version:     1,
-				Name:        "fixed:datasources:compatibility:querier",
-				Description: "Query data sources when data source permissions are not in use",
-				Permissions: []accesscontrol.Permission{
-					{Action: ActionDatasourcesQuery},
+		Grants: []string{string(models.ROLE_ADMIN)},
+	}
+
+	datasourcesIdReaderRole := accesscontrol.RoleRegistration{
+		Role: accesscontrol.RoleDTO{
+			Version:     3,
+			Name:        "fixed:datasources.id:reader",
+			DisplayName: "Data source ID reader",
+			Description: "Read the ID of a data source based on its name.",
+			Permissions: []accesscontrol.Permission{
+				{
+					Action: ActionDatasourcesIDRead,
+					Scope:  ScopeDatasourcesAll,
 				},
 			},
-			Grants: []string{string(models.ROLE_VIEWER)},
 		},
-		{
-			Role: accesscontrol.RoleDTO{
-				Version:     2,
-				Name:        "fixed:current:org:reader",
-				Description: "Read current organization and its quotas.",
-				Permissions: []accesscontrol.Permission{
-					{
-						Action: ActionOrgsRead,
-					},
-					{
-						Action: ActionOrgsQuotasRead,
-					},
-				},
+		Grants: []string{string(models.ROLE_VIEWER)},
+	}
+
+	datasourcesCompatibilityReaderRole := accesscontrol.RoleRegistration{
+		Role: accesscontrol.RoleDTO{
+			Version:     2,
+			Name:        "fixed:datasources:compatibility:querier",
+			DisplayName: "Data source compatibility querier",
+			Description: "Query data sources when data source permissions are not in use",
```

Comment:
```suggestion
			Description: "Query data sources when data source permissions are not in use.",
```

Reply:
I changed to use the description we put in the rename doc